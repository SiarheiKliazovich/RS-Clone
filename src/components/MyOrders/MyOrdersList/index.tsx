import { FunctionComponent } from "react";
import { useTranslation } from "react-i18next";

import { IOrdersListComponent } from "../../../interfaces/ordersListComponent";

const MyOrdersList: FunctionComponent<IOrdersListComponent> = ({ orders }) => {
  const { t } = useTranslation("dataLang");

  return (
    <div className="market-block__products">
      <div className="row__products">
        {orders.map((item, i) => {
          return (
            <div key={i} className="tie__product my-ties">
              <div className="tie__products_img">
                <div>
                  <img className="products_img" src={item.image} alt="" />
                </div>
              </div>
              <div className="tie__products_discription my-orders-list__item">
                <p className="tie__products_name">{t("myOrders.data")}</p>
                <div className="item__date">
                  {t("myOrders.date")}: {item.date.slice(0, 10)}
                </div>
                <div className="item__price">
                  {t("myOrders.sum")}: {item.price}$
                </div>
                <div className="item__status">
                  {t("myOrders.status")}: {item.status}
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default MyOrdersList;
