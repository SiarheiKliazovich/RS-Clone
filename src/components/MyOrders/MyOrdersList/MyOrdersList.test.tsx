import { render } from "@testing-library/react";
import MyOrdersList from ".";

describe("MyOrdersList", () => {
  test("renders correctly", () => {
    const orders = [
      {
        _id: "1",
        image: "image-url-1",
        date: "2022-01-01T00:00:00.000Z",
        price: 100,
        status: "completed",
        userId: "user",
        sellerId: "seller",
      },
      {
        _id: "2",
        image: "image-url-2",
        date: "2022-01-02T00:00:00.000Z",
        price: 200,
        status: "in progress",
        userId: "user",
        sellerId: "seller",
      },
    ];
    const { container } = render(<MyOrdersList orders={orders} />);
    expect(container).toMatchSnapshot();
  });
});
