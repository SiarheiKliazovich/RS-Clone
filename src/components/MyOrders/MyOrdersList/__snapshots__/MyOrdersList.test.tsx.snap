// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyOrdersList renders correctly 1`] = `
<div>
  <div
    class="market-block__products"
  >
    <div
      class="row__products"
    >
      <div
        class="tie__product"
      >
        <div
          class="tie__products_img"
        >
          <div>
            <img
              alt=""
              class="products_img"
              src="image-url-1"
            />
          </div>
        </div>
        <div
          class="tie__products_discription"
        >
          <p
            class="tie__products_name"
          >
            myOrders.data
          </p>
          <div
            class="item__date"
          >
            myOrders.date
            : 
            2022-01-01
          </div>
          <div
            class="tie__products_price"
          >
            myOrders.sum
            : 
            100
            $
          </div>
          <div
            class="item__status"
          >
            myOrders.status
            : 
            completed
          </div>
        </div>
      </div>
      <div
        class="tie__product"
      >
        <div
          class="tie__products_img"
        >
          <div>
            <img
              alt=""
              class="products_img"
              src="image-url-2"
            />
          </div>
        </div>
        <div
          class="tie__products_discription"
        >
          <p
            class="tie__products_name"
          >
            myOrders.data
          </p>
          <div
            class="item__date"
          >
            myOrders.date
            : 
            2022-01-02
          </div>
          <div
            class="tie__products_price"
          >
            myOrders.sum
            : 
            200
            $
          </div>
          <div
            class="item__status"
          >
            myOrders.status
            : 
            in progress
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
